modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -2271,6 +2296,25 @@ private void ackNodeMetrics(DecimalFormat dblFmt,
         }
     }
 
+    private long allocatedPDSSize() {
+        DataStorageConfiguration dsCfg = ctx.config().getDataStorageConfiguration();
+
+        if (dsCfg == null)
```

Comment:
Based on the IgnitionEx#initializeDataStorageConfiguration(cfg) implementation the DataStorageConfiguration can't be null as long as it's not a client node. Should we add "ctx.clientNode()" check here?

Reply:
added client node check